rust/core-lib/src/event_context.rs

```diff
@@ -839,4 +839,145 @@ mod tests {
+        assert_eq!(harness.debug_render(),"this is a 0| text example");
+
+        // as long as the beginning of a region starts on a number,
+        // conversion will happen
+        ctx.do_edit(EditNotification::MoveToEndOfDocument);
+        ctx.do_edit(EditNotification::DeleteToBeginningOfLine);
+        ctx.do_edit(EditNotification::Insert { chars: "this is a 10 text example".into() });
+        ctx.do_edit(EditNotification::Gesture { line: 0, col: 11, ty: PointSelect });
+        ctx.do_edit(EditNotification::MoveToEndOfDocumentAndModifySelection);
+        ctx.do_edit(EditNotification::DecreaseNumber);
+        assert_eq!(harness.debug_render(),"this is a 9[ text example|]");
+
+        // If a number just happens to be in a region, nothing will happen
+        ctx.do_edit(EditNotification::MoveToEndOfDocument);
+        ctx.do_edit(EditNotification::DeleteToBeginningOfLine);
+        ctx.do_edit(EditNotification::Insert { chars: "this is a 10 text example".into() });
+        ctx.do_edit(EditNotification::Gesture { line: 0, col: 5, ty: PointSelect });
+        ctx.do_edit(EditNotification::MoveToEndOfDocumentAndModifySelection);
+        ctx.do_edit(EditNotification::DecreaseNumber);
+        assert_eq!(harness.debug_render(),"this [is a 10 text example|]");
+
+        // if it ends on a region, nothing will happen
+        ctx.do_edit(EditNotification::MoveToEndOfDocument);
+        ctx.do_edit(EditNotification::DeleteToBeginningOfLine);
+        ctx.do_edit(EditNotification::Insert { chars: "this is a 10".into() });
+        ctx.do_edit(EditNotification::Gesture { line: 0, col: 0, ty: PointSelect });
+        ctx.do_edit(EditNotification::MoveToEndOfDocumentAndModifySelection);
+        ctx.do_edit(EditNotification::DecreaseNumber);
+        assert_eq!(harness.debug_render(),"[this is a 10|]");
+
+        // if only a part of a number is in a region, the whole number will be changed
+        ctx.do_edit(EditNotification::MoveToEndOfDocument);
+        ctx.do_edit(EditNotification::DeleteToBeginningOfLine);
+        ctx.do_edit(EditNotification::Insert { chars: "this is a 1000 text example".into() });
+        ctx.do_edit(EditNotification::Gesture { line: 0, col: 11, ty: PointSelect });
+        ctx.do_edit(EditNotification::MoveRightAndModifySelection);
+        ctx.do_edit(EditNotification::DecreaseNumber);
+        assert_eq!(harness.debug_render(),"this is a 999| text example");
+
+        // Number Wrapping
+        ctx.do_edit(EditNotification::MoveToEndOfDocument);
+        ctx.do_edit(EditNotification::DeleteToBeginningOfLine);
+        ctx.do_edit(EditNotification::Insert { chars: format!("{}", i128::max_value()) });
+        ctx.do_edit(EditNotification::Gesture { line: 0, col: 11, ty: PointSelect });
+        ctx.do_edit(EditNotification::MoveRightAndModifySelection);
+        ctx.do_edit(EditNotification::IncreaseNumber);
+        assert_eq!(harness.debug_render(), format!("{}|", i128::min_value()));
```

Comment:
with the assert macros you don't need the `format!` call; the asserts treat all arguments after the statement as a format argument, so you can just write, say, `assert!(false, "this is a {} example", "pretty bad");`